Got up to here on weekend challenge: RPS

BenjaminNeustadt opened the pull request.

BenjaminNeustadt committed:
started draft on Rock paper scissor challenge

BenjaminNeustadt committed:
Module RPS instead of class

We can now grab the rules of the Rock Paper Scissors game and use it in
our game, or at the "main" or "global" space.

Currently, this is at the global level.

BenjaminNeustadt committed:
Mac use is a personal choice.

We should not have personal choice things in the project gitignore.
They belong in the developers personal "global" excludes file instead.

raphaella-rose commented on spec/features/player_spec.rb:
```diff
@@ -0,0 +1,12 @@
+  subject(:player) { Player.new('Player') }
+  subject(:computer) { Player.new('Computer') }
+
+  describe '#name' do
```
This is a good initial test for the player class

raphaella-rose commented on views/name.erb:
```diff
@@ -0,0 +1,6 @@
+We will now ask you for your name.
```
This is straight forward and easy to understand :)

raphaella-rose commented on controller.rb:
```diff
@@ -0,0 +1,47 @@
+  end
+
+  post '/assign_name' do
+    @player = Player.new(params[:name])
```
You could use a session to store the player name

raphaella-rose reviewed (commented):
Using more tests to drive your code may help you know which direction to go in. I thought your code was easy to read. :)

BenjaminNeustadt committed:
User story 1: added config.ru file and ENV[RACK_ENV] = test

require File.join(File.dirname(__FILE__), .., app.rb) in the spec_helper:
rspec now passing correctly. enabled session in controller, stored instance variable in controller
[started again with tests this time]

User Story 1

As a marketeer
So that I can see my name in lights
I would like to register my name before playing an online game

BenjaminNeustadt committed:
User Story 2

As a marketeer
So I can play a game
I want to see the shapes I can play

made minimum adjustments necessary to pass rspec test. hardcoded the
words "Rock" "Paper" "Scissors" to the erb files.

BenjaminNeustadt committed:
User Story 3

 # As a marketeer
  # So I can play a game
  # I want to choose a shape to play

created instance variable @shape and used sessions to store it with
params[:shape].

This is possible because in the <form> we give to
name="shape". Params are passed in the form of hash. Provide a way for
information to be passed from a user to your application. Params are
used to allow interactivity on a website, for example by allowing form
data to be sent.

BenjaminNeustadt committed:
User Story 4

As a marketeer
So I can play a game
I want the game to choose an option

FUN FACT: normal methods can be written into the test file.Capybara has
methods not only to match something on the page but also retrieve element from page and see
the contents of the element.CF method inside the play_spec file.

BenjaminNeustadt committed:
User Story 5

As a marketeer
So I can play a game
I want the game to choose a random option

Used srand to get pseudo random number that we put into a seed at the
beginning in a constant.

BenjaminNeustadt committed:
Refactored controller to pass only one instance variable to the view.
For this created an object Turn, and used options on that object to
refer to different types of objects within that.

BenjaminNeustadt committed:
User Story 5: declare a winner

End Game.
Implemented logic in model

BenjaminNeustadt committed:
Final adjustments

Added necessary 'require' for the models held in lib folder.
